### PR TITLE
Resimulate transaction on revert during waiting for confirmation to get error message

### PR DESCRIPTION
**Pull request by yoshiwarab**
https://linear.app/art-blocks/issue/CUR-1837/change-error-when-topupbid-is-reverted

### CODE REVIEW

**Review by Asupkay — APPROVED** [review]

**Comment by yoshiwarab** [comment]
@ryley-o I think this is the only way we can get the revert message but let me know if you know of something more direct!

**Review by asaltste — APPROVED** [review]

**Review by ryley-o — APPROVED** [review]
there might be something more direct with `getTransactionError`, and re-simulating could result in a different error here, but I think this is good in the short term

File: packages/sdk/src/machines/ram-machine/index.ts [hunk 1]
```diff
@@ -303,17 +305,38 @@ export const ramMachine = setup({
...
+          // been updated between the receipt being returned and the trace
+          // being fetched it is possible that the revert reason will not be
+          // accurate in this case.
+          try {
+            const traceResult = await traceTransaction(publicClient, txHash);
+
+            throw new ContractFunctionRevertedError({
+              abi: minterRAMV0Abi,
+              functionName: txRequest.functionName,
+              data: traceResult?.[0]?.result?.output,
```
Comment (ryley-o) [inline packages/sdk/src/machines/ram-machine/index.ts:323]: Just to check, does this get properly decoded on the frontend? I think the revert reason is in args[0], but maybe we are accessing that when decoding later on?

**Review by ryley-o — APPROVED** [review]
LGTM - have you validated this is wai?

**Pull request merged**